setup/features/user-data.md

```diff
@@ -0,0 +1,143 @@
+
+The filename and location of the file is configured differently for each provider.
+The general format for the template file a simple property file though there are some differences depending on the provider.
+
+Example of setting a Consul server address:
+
+```bash
+CONSUL_SERVER_ADDRESS="consul.example.com"
+```
+
+The environment variable `CONSUL_SERVER_ADDRESS` will be set on every instance with the value of `consul.example.com`.
+
+Typical usage would be defining environment variables with the values including the environment specific details
+to customize the behavior based on account, environment, region, and other things.
+
+Here is the list of replacement tokens:
+
+**Token**          | **Description**
+------------------ | ----------------------------------------
+`%%account%%`      | Name of the account
+`%%accounttype%%`  | Type of the account
+`%%env%%`          | Environment of the account
+`%%app%%`          | Application being deployed
+`%%region%%`       | The deployment region
+`%%group%%`        | Name of the server group
+`%%cluster%%`      | Name of the cluster
+`%%stack%%`        | Stack component of the cluster name
+`%%detail%%`       | Detail component of the cluster name
+`%%launchconfig%%` | Name of the launch configuration
+
+Example template file using tokens in the values:
+
+```bash
+CLOUD_ACCOUNT="%%account%%"
+CLOUD_ACCOUNT_TYPE="%%accounttype%%"
+CLOUD_ENVIRONMENT="%%env%%"
+CLOUD_SERVER_GROUP="%%group%%"
+CLOUD_CLUSTER="%%cluster%%"
+CLOUD_STACK="%%stack%%"
+CLOUD_DETAIL="%%detail%%"
+CLOUD_REGION="%%region%%"
+```
+
+If the server group udf-example-cluster-v001 was deployed using this template in the account `main`, accountType `streaming`, environment `prod`, in the `east` region, the resulting user data would look like:
+
+```bash
+CLOUD_ACCOUNT="main"
+CLOUD_ACCOUNT_TYPE="streaming"
+CLOUD_ENVIRONMENT="prod"
+CLOUD_SERVER_GROUP="udf-example-cluster-v001"
+CLOUD_CLUSTER="udf-example-cluster"
+CLOUD_STACK="example"
+CLOUD_DETAIL="cluster"
+CLOUD_REGION="east"
+```
+
+### AWS
+
+The location of the template file for AWS is controlled by the `udf.udfRoot` property and the behavior is controlled by the `udf.defaultLegacyUdf` property. The defaults are:
+
+````yaml
+udf:
+  udfRoot: /apps/nflx-udf
+  defaultLegacyUdf: true
+````
+
+You probably want to change the location on the filesystem where the template file lives to suit your deployment.
+You almost certainly want to change `udf.defaultLegacyUdf=false` as this disables deprecated behavior that is specific to Netflix.
+In the `udf.udfRoot` directory, create a file called `udf0`. The contents of this file are base64 encoded and set as the `user-data` for the LaunchConfiguration when a new server group is created.
+
+**Note**: The `udf` property is root level property instead of being nested under the AWS section.
+
+### Google
+
+The location of the template file for Google is controlled by the `google.accounts.userDataFile`
+
+```yaml
+google:
```

Comment:
Please replace with the equivalent halyard command

Reply:
I think I figured it out. The Halyard docs are pretty good :)